jinahub/rankers/CatboostRanker/catboost_ranker.py

```diff
@@ -97,33 +97,36 @@ def build_catboost_pool(self, docs: DocumentArray):
             return Pool(data=data, label=label, group_id=group_id)
 
     @requests(on='/train')
-    def train(self, docs: DocumentArray, parameters: Dict, **kwargs):
+    def train(self, docs: DocumentArray, parameters: Dict = {}, **kwargs):
         catboost_parameters = parameters.get(
             'catboost_parameters', self.catboost_parameters
         )
         train_pool = self.build_catboost_pool(docs)
         self.model = CatBoostRanker(**catboost_parameters)
         self.model.fit(train_pool)
 
-    @requests(on='/predict')
-    def predict(self, docs: DocumentArray, parameters: Dict, **kwargs):
-        catboost_parameters = parameters.get(
-            'catboost_parameters', self.catboost_parameters
-        )
-        predict_pool = self._extract_features_from_docs(docs)
-        self.model = CatBoostRanker(**catboost_parameters)
-        self.model.fit(predict_pool)
+    @requests(on='/rank')
+    def rank(self, docs: DocumentArray, parameters: Dict = {}, **kwargs):
+        feature_vecs, _, _ = self._extract_features(docs)
+        if not self.model.is_fitted():
+            raise ValueError(
+                'You need to train your model before make prediction, call `train` endpoint first.'
+            )
+        preds = self.model.predict(feature_vecs)
+        matches = docs.traverse_flat(traversal_paths=['m'])
+        for match, pred in zip(matches, preds):
```

Comment:
The objective of the ranker is to change the order of the matches which is not happening here. What catboost ranker should give is a score